### DISCUSSION TITLE
Authentication Error

### DISCUSSION THREAD

**Discussion by michaelturner008** [saleweaver/python-amazon-sp-api#88] [Q&A] [answered]
I'm new to the SP api, altogether, and I'm having trouble authenticating my calls.  
Having set up a developer account, walked through the application registration process in Amazon's [documentation](https://github.com/amzn/selling-partner-api-docs/blob/main/guides/developer-guide/SellingPartnerApiDeveloperGuide.md#registering-your-selling-partner-api-application), I'm still struggling to authenticate. 

My code is  

---------

from sp_api.api import Reports

credentials= dict(
    refresh_token='XXXXXXXX',
    lwa_client_secret='XXXXXXXX',
    aws_secret_key='XXXXXXXX',
    aws_access_key='XXXXXXXX',
    role_arn='XXXXXXXX',
)

createReportResponse = Reports(credentials = credentials).create_report(reportType='GET_FLAT_FILE_OPEN_LISTINGS_DATA')`

--------

I'm receiving the following error: sp_api.auth.exceptions.AuthorizationError: ('invalid_client', 'Client authentication failed', 401)

Is there any way to get further understanding into what portion of the authentication is failing? I've grabbed each of that params for their respective places (developer central & AWS console), so it seems as though everything should be accurate. 

**EDIT:** I have re-followed the setup instructions step by step two separate times to ensure all the credentials above are correct, and am still experiencing the same error.

Any help would be greatly appreciated.

**Comment by chesterSF** [comment]
I'm getting the same error and stuck as well.

I Googled around and made sure I was supplying role_arn as per instruction.

Did you self-authorize your app as well? I wonder if that could be the cause related to this thread https://github.com/amzn/selling-partner-api-docs/issues/66

Does anyone know? Very much appreciated!

**Comment by saleweaver** [comment, marked as answer]
Did you try other endpoints?

**Reply by chesterSF** [reply]
Hi saleweaver,


Thanks! Yes.  I tried 

resp = Orders(credentials=credentials, marketplace=Marketplaces.US).get_orders(CreatedAfter=(datetime.utcnow() - timedelta(days=7)).isoformat())

The creds look like this:

marketpalce_id = 'ATVPDKIKX0DER'

app_id = 'amzn1.sp.solution.xxxx....'
refresh_token = 'yyyxxx.... very long'
lwa_client_id = 'amzn1.application-oa2-client.xxxx...'
lwa_client_secret = 'xxxyyxz....'

aws_access_key = 'AKIxxxxx'
aws_secret_key = 'yyyyy'
role_arn = 'arn:aws:iam::#####:role/SellingPartnerAPIRole'

credentials = dict(
        refresh_token=refresh_token,  # Seller Central Dev Central Action/Authorize after account creation
        lwa_app_id=app_id,  # Seller Central
        lwa_client_secret=lwa_client_secret,
        aws_secret_key=aws_secret_key,
        aws_access_key=aws_access_key,
        role_arn=role_arn,
    )

**Reply by chesterSF** [reply]
ianrussel in another thread answered this question.
I'll post the answer here just in case it helps others.

It turns out I had to set 

app_id=lwa_client_id (from View LWA credentials)
Instead of app_id from Seller Central's "App ID" column in the table (as I did in my code above).
